tasks/request_from_archive/request_from_archive.py

```diff
@@ -167,51 +168,43 @@ def get_glacier_recovery_type(config: Dict[str, Any]) -> str:
     # Look for config override
     recovery_type = config.get(CONFIG_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY, None)
     if recovery_type is not None:
-        if recovery_type in VALID_RESTORE_TYPES:
+        LOGGER.info(
+            f"Using restore type of {recovery_type} "
+            f"found in the configuration {CONFIG_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY} key."
+        )
+    else:
+        # Look for default from TF
+        recovery_type = os.getenv(OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY, None)
+        if recovery_type is not None:
             LOGGER.info(
                 f"Using restore type of {recovery_type} "
-                f"found in the configuration {CONFIG_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY} key."
+                f"found in the environment {OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY} key."
             )
         else:
-            LOGGER.error(
-                f"Invalid restore type value of '{recovery_type}' "
-                f"found in the configuration {CONFIG_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY} key."
-            )
-            raise ValueError(
-                f"Invalid restore type value in configuration "
-                f"{CONFIG_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY} key."
+            recovery_type = "Standard"
+            LOGGER.info(
+                f"Using restore type of {recovery_type} "
+                f"due to lack of overrides."
             )
 
-    # Look for default from TF
-    if recovery_type is None:
-        recovery_type = os.getenv(OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY, None)
-        if recovery_type is not None:
-            if recovery_type in VALID_RESTORE_TYPES:
-                LOGGER.info(
-                    f"Using restore type of {recovery_type} "
-                    f"found in environment variable {OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY}."
-                )
-            else:
-                LOGGER.error(
-                    f"Invalid restore type value of '{recovery_type}' "
-                    f"found in environment variable {OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY}."
-                )
-                raise ValueError(
-                    f"Invalid restore type value in environment variable "
-                    f"{OS_ENVIRON_DEFAULT_RECOVERY_TYPE_KEY}"
-                )
-
+    if recovery_type not in VALID_RESTORE_TYPES:
```

Comment:
Moved ValueError check to end. Will now need to be combined with info logging to fully explain error, but reduces complexity.

Reply:
I think this is smart since we only need to validate that the value is correct once when we choose where we get the value from override or default.